src/mongo/shell.ts

```diff
@@ -54,9 +57,9 @@ export class Shell {
 		let buffers: string[] = [];
 		on(this.mongoShell.stdout, 'data', b => {
 			let data: string = b.toString();
-			const delimitter = `${this.executionId}${os.EOL}`;
-			if (data.endsWith(delimitter)) {
-				const result = buffers.join('') + data.substring(0, data.length - delimitter.length);
+			const endOfDataSentinel = `${endOfDataBaseSentinel}${this.executionId}${os.EOL}`;
+			if (data.endsWith(endOfDataSentinel)) {
```

Comment:
I'm fine with this change, but we honestly need to completely refactor this file at some point (this code was inherited from the old mongo repo). If we copy our [cpUtils code](https://github.com/Microsoft/vscode-azurefunctions/blob/master/src/utils/cpUtils.ts#L33) from another repo I think we can completely avoid the use of a sentinel. I might take a look after the upcoming Remote bug-fix release.

Reply:
Um, the refactor already has a PR.

Reply:
I just wanted this separate because it was specifically related to a bug and I did it before the rewrite.

Reply:
Sorry for missing that lol - filtered out a few "Post build" emails last week in the lead up to build